### PR TITLE
Collective indexing merged

### PR DESCRIPTION
**Pull request by tisto**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -48,7 +48,8 @@
         'Products.GenericSetup',
         'ZODB3',
         'Zope2 >= 2.13',
-        'collective.indexing >= 2.0a2',
```
Comment (tisto) [inline setup.py]: @gforcada removing collective.indexing here is not a good approach in my opinion. This means that all users of Plone < 5.1 have to add collective.indexing as an explicit dependency.

File: src/collective/solr/interfaces.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-from collective.indexing.interfaces import IIndexQueueProcessor
+from Products.CMFCore.interfaces import IIndexQueueProcessor
```
Comment (tisto) [inline src/collective/solr/interfaces.py]: @gforcada this breaks collective.solr for all Plone versions < 5.1.

**Comment by tisto** [comment]
@gforcada I tried to re-visit this and make the collective.indexing dependency optional. You never came back to my comments on your PR, so I have no idea about your opinion regarding that topic.

Though, when I try to find where things moved after the c.indexing merge (e.g. getOwnIndexMethod from collective.indexing.indexer) I don't find them in the merges and I don't know where to look TBH. I wasn't involved in the PLIP and I never reviewed any of the code.

Would you be willing to help on this? Otherwise my best bet would be trying to migrate collective.indexing to work with Plone 5.2/Python 3.

**Comment by gforcada** [comment]
The PR and the comments on it that you refer to is this one: https://github.com/collective/collective.solr/pull/206 ?

You were asking me to make collective.indexing optional rather than removing it completely, but at the same time you were the one not sure if I should push that to a non-existing branch, and that there are tests to be fixed on older releases and that you will make a release, but that there are things to fix first there as well...

So yeah, I did not reply, I got that as a: *thanks for working on this, but don't push it now, because I have other priorities and this would just get on the way*.

As for this commit on this pull request, which the commit message already quite clearly states its status (WIP), it was mostly a quick try and tests to see if there would be a lot of breakage. On #206 there is far more work (multiple work days on it actually) and you actually create a pull request out of a WIP commit of mine, rather than the other pull request... I don't get it really.

We, at der Freitag, will try to migrate to python 3 before 2020, most likely like everyone else :sweat_smile: , that will of course involve collective.solr. If we can do it upstream or not, seems to not be for us to decide.

**Comment by tisto** [comment]
Closing in favor of https://github.com/collective/collective.solr/pull/222

**Pull request closed without merging**